### PR TITLE
remove hardcoded yaml-cpp path from CMakeLists.txt

### LINKED ISSUE #6: cmake contains hardcoded paths for yaml-cpp library
`CMakeLists.txt` and `examples/cpp/cart_pole/CMakeLists.txt` contain hardcoded paths to `yaml-cpp`

### PR DESCRIPTION
**Pull request by TomMelt**
`CMakeLists.txt` and `examples/cpp/cart_pole/CMakeLists.txt` contain hardcoded paths to `yaml-cpp`

These has been removed in favour of using `env` variables

* `YAML_LIBRARY`
* `YAML_INCLUDE_DIRS`

Fixes #6

### CODE REVIEW

**Comment by azrael417** [comment]
LGTM. Josh, any comments?

File: examples/cpp/cart_pole/CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,7 @@ target_link_libraries(PyEnvironments PRIVATE environments)
 foreach(tgt ${cart_pole_example_targets})
   target_include_directories(${tgt}
     PRIVATE
-    /opt/yaml-cpp/include
+    ${YAML_INCLUDE_DIRS}
```
Comment (romerojosh) [inline examples/cpp/cart_pole/CMakeLists.txt:25]:
```suggestion
    ${YAML_CPP_ROOT}/include
```

File: examples/cpp/cart_pole/CMakeLists.txt [hunk 2]
```diff
@@ -32,7 +32,7 @@ foreach(tgt ${cart_pole_example_targets})
   target_link_libraries(${tgt} PRIVATE ${TORCH_LIBRARIES})
   target_link_libraries(${tgt} PRIVATE ${Python_LIBRARIES})
   target_link_libraries(${tgt} PRIVATE NVHPC::MPI_CXX)
-  target_link_libraries(${tgt} PRIVATE /opt/yaml-cpp/lib/libyaml-cpp.a)
+  target_link_libraries(${tgt} PRIVATE ${YAML_LIBRARY}/libyaml-cpp.a)
```
Comment (romerojosh) [inline examples/cpp/cart_pole/CMakeLists.txt:35]:
```suggestion
  target_link_libraries(${tgt} PRIVATE ${YAML_CPP_ROOT}/lib/libyaml-cpp.a)
```

**Review by romerojosh — CHANGES_REQUESTED** [review]
Thanks for working on improving the build for non-Docker users @TomMelt!

Left a couple of comments/changes. Additionally, before we can merge, can you update the `Dockerfile` to include the definition of this new env var to the TorchFort build line, for example, something like:
```
RUN cd /torchfort && mkdir build && cd build && \
    cmake -DCMAKE_INSTALL_PREFIX=/opt/torchfort \
    -DNVHPC_CUDA_VERSION=11.8 \
    -DCMAKE_PREFIX_PATH="`python -c 'import torch;print(torch.utils.cmake_prefix_path)'`;/opt/nvidia/hpc_sdk/Linux_x86_64/23.5/cmake" \
    -DYAML_LIBRARY=/opt/yaml-cpp/lib \
    -DYAML_INCLUDE_DIRS=/opt/yaml-cpp/include \
    .. && \
    make -j install && \
    cd / && rm -rf torchfort
```

**Comment by TomMelt** [comment]
> Thanks for working on improving the build for non-Docker users @TomMelt!
> 
> Left a couple of comments/changes. Additionally, before we can merge, can you update the `Dockerfile` to include the definition of this new env var to the TorchFort build line, for example, something like:
> 
> ```
> RUN cd /torchfort && mkdir build && cd build && \
>     cmake -DCMAKE_INSTALL_PREFIX=/opt/torchfort \
>     -DNVHPC_CUDA_VERSION=11.8 \
>     -DCMAKE_PREFIX_PATH="`python -c 'import torch;print(torch.utils.cmake_prefix_path)'`;/opt/nvidia/hpc_sdk/Linux_x86_64/23.5/cmake" \
>     -DYAML_LIBRARY=/opt/yaml-cpp/lib \
>     -DYAML_INCLUDE_DIRS=/opt/yaml-cpp/include \
>     .. && \
>     make -j install && \
>     cd / && rm -rf torchfort
> ```

Hi @romerojosh , I have update the `Dockerfile` to include the `YAML_CPP_ROOT` environment variable. It should be sufficient.

https://github.com/TomMelt/TorchFort/blob/218ac7de1c0b609a18cae6353ef72c4baeb88f89/docker/Dockerfile#L48

**Review by romerojosh — APPROVED** [review]
LGTM! Thanks for the PR!

**Pull request merged**